padertorch/summary/tbx_utils.py

```diff
@@ -68,6 +75,54 @@ def stft_to_image(signal, batch_first=False, color='viridis'):
 _spectrogram_to_image_cmap = {}
 
 
+def _colorize(image, color):
+    """
+    >>> i = np.arange(15).reshape([3, 5])
+    >>> _colorize(i, True).shape
+    (4, 3, 5)
+    >>> _colorize(i, None).shape
+    (1, 3, 5)
+    >>> i = np.arange(6).reshape([2, 3])
+    >>> _colorize(i, True)
+    array([[[0.267004, 0.26851 , 0.269944],
+            [0.271305, 0.272594, 0.273809]],
+    <BLANKLINE>
+           [[0.004874, 0.009605, 0.014625],
+            [0.019942, 0.025563, 0.031497]],
+    <BLANKLINE>
+           [[0.329415, 0.335427, 0.341379],
+            [0.347269, 0.353093, 0.358853]],
+    <BLANKLINE>
+           [[1.      , 1.      , 1.      ],
+            [1.      , 1.      , 1.      ]]])
+    >>> _colorize(i, None)
+    array([[[0, 1, 2],
+            [3, 4, 5]]])
+    """
+    if color is None:
+        return image[None, :, :]
+    else:
+        if color is True:
+            color = 'viridis'
+        try:
+            cmap = _spectrogram_to_image_cmap[color]
+        except KeyError:
```

Comment:
since we are using this exact code already at two places it should be put in a separate function

Reply:
It is already a separate function. I removed the old code and replaced it with this function